### PR TITLE
ilc with service call to yield

### PR DESCRIPTION
**Pull request by saina-ramyar**
# PR Details
## Description
Communication between ILC and Yield plygin, with service call to yield, from inside the ILC trajectory callback function.

This PR is just for comparison purposes and is not meant to be merged with platform.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: inlanecruising_plugin/config/parameters.yaml [hunk 1]
```diff
@@ -7,4 +7,4 @@ moving_average_window_size: 5 # Size of the window used in the moving average fi
 back_distance: 15.0 # Number of meters behind the first maneuver that need to be included in points for curvature calculation
 max_accel_multiplier: 0.85 # Multiplier of max_accel to bring the value under max_accel
 lat_accel_multiplier: 0.50 # Multiplier of lat_accel to bring the value under lat_accel TODO needs to be optimized
-enable_object_avoidance: false # Activates object avoidance logic 
\ No newline at end of file
+enable_object_avoidance: true # Activates object avoidance logic
```
Comment (kjrush) [inline inlanecruising_plugin/config/parameters.yaml:10]: Let's default this to off until we've finished integration testing on this feature.

File: inlanecruising_plugin/include/inlanecruising_plugin/inlanecruising_plugin.h [hunk 2]
```diff
@@ -72,6 +72,9 @@ class InLaneCruisingPlugin
    */ 
   bool plan_trajectory_cb(cav_srvs::PlanTrajectoryRequest& req, cav_srvs::PlanTrajectoryResponse& resp);
 
+  bool plan_trajectory_cb2(cav_srvs::PlanTrajectoryRequest& req, cav_srvs::PlanTrajectoryResponse& resp);
```
Comment (kjrush) [inline inlanecruising_plugin/include/inlanecruising_plugin/inlanecruising_plugin.h:75]: Would be nice to have doc comments for these functions.

File: inlanecruising_plugin/src/helper.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#include "ros/ros.h"
```
Comment (kjrush) [inline inlanecruising_plugin/src/helper.cpp:1]: Copyright statement

File: inlanecruising_plugin/src/inlanecruising_plugin.cpp [hunk 4]
```diff
@@ -60,10 +60,53 @@ bool InLaneCruisingPlugin::onSpin()
...
+void InLaneCruisingPlugin::set_yield_client(ros::ServiceClient& client)
+{
+  yield_client_ = client;
+}
+
+bool InLaneCruisingPlugin::plan_trajectory_cb2(cav_srvs::PlanTrajectoryRequest& req,
+                                              cav_srvs::PlanTrajectoryResponse& resp)
+{
+
+  ROS_ERROR("in cb2");
```
Comment (kjrush) [inline inlanecruising_plugin/src/inlanecruising_plugin.cpp:72]: Is this actually an error?

File: inlanecruising_plugin/src/inlanecruising_plugin.cpp [hunk 5]
```diff
@@ -60,10 +60,53 @@ bool InLaneCruisingPlugin::onSpin()
...
+  if (yield_client_ && yield_client_.exists() && yield_client_.isValid()){
+    ROS_ERROR("Good Client");
+  }
+  cav_srvs::PlanTrajectory yield_srv;
+  if (req.initial_trajectory_plan.trajectory_id == "ILCReq"){
```
Comment (kjrush) [inline inlanecruising_plugin/src/inlanecruising_plugin.cpp:77]: Just checking, are these `trajectory_id`s with specific values only used for the unit testing? Nominally I believe our practice here is to use randomly-assigned GUIDs for trajectory ID's for traceability.

File: inlanecruising_plugin/src/inlanecruising_plugin.cpp (hunk 4 above)
Comment (kjrush) [reply inlanecruising_plugin/src/inlanecruising_plugin.cpp:72]: If this method is just for the unit/integration testing, I suppose this doesn't matter too much

File: inlanecruising_plugin/src/inlanecruising_plugin.cpp [hunk 6]
```diff
@@ -60,10 +60,53 @@ bool InLaneCruisingPlugin::onSpin()
...
+      {
+        ROS_ERROR("Failed to call service ");
+      } 
+  }
+    else ROS_ERROR("Service Unavailable");
+  
+  return true;
+}
+
 bool InLaneCruisingPlugin::plan_trajectory_cb(cav_srvs::PlanTrajectoryRequest& req,
```
Comment (kjrush) [inline inlanecruising_plugin/src/inlanecruising_plugin.cpp:106]: Should I be expecting to see the yield plugin invoked here? Or am I missing where that happens?

File: inlanecruising_plugin/config/parameters.yaml (hunk 1 above)
Comment (kjrush) [reply inlanecruising_plugin/config/parameters.yaml:10]: Does this param get referenced anywhere anymore? It looks like the only usage of it (in `plan_trajectory_cb`) was removed.

**Comment by msmcconnell** [comment]
@saina-ramyar @kjrush Is this PR still valid? It's been open since Febuary

**Comment by saina-ramyar** [comment]
This PR is not valid. It was set up for discussion only. Closing.

**Pull request closed without merging**